Packs/Legacy/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py

```diff
@@ -326,27 +342,34 @@ def main():
                                                                              raise_error=False),
                                                     SIMILAR_INCIDENTS_FIELDS_MAP)
                                ]
-    # filter by context
-    if incident_similar_context:
-        filter_by_context = []
+    filter_by_context = []
+    if original_context_map:
         for c in duplicate_incidents:
-            other_context = get_map_from_nested_dict(get_context(c['id']), SIMILAR_CONTEXT_MAP.keys())
-            if other_context:
-                if verify_map_equals(other_context,
-                                     incident_similar_context,
-                                     SIMILAR_CONTEXT_MAP):
-                    filter_by_context.append(c)
+            other_incident_context = get_context(c['id'])
+
+            if other_incident_context:
+                other_incident_context_map = {}
+
+                for key in SIMILAR_CONTEXT_MAP.keys():
+                    response = demisto.dt(other_incident_context, key)
+                    if response:
+                        other_incident_context_map[key] = response
+                        if verify_map_equals(original_context_map, other_incident_context_map, SIMILAR_CONTEXT_MAP):
+                            filter_by_context.append(c)
         duplicate_incidents = filter_by_context
```

Comment:
Why do you need both `filter_by_context` and `duplicate_incidents`

Reply:
the for loop is on duplicate_incidents, so we cant append to it and that is why we need a different list